pymc_extras/statespace/models/structural/core.py

```diff
@@ -540,6 +565,8 @@ def __init__(
 
         self.param_info = {}
         self.data_info = {}
+        self.shock_info = {}
+        self.state_info = {}
```

Comment:
Lets initialize these with empty versions of the relevant dataclasses, or `None` if you prefer. They should never be a dictionary.